src/main/resources/META-INF/plugin.xml

```diff
@@ -18,15 +19,19 @@
         <changesViewContent tabName="Git Machete"
                             className="com.virtuslab.gitmachete.frontend.ui.impl.root.GitMacheteContentProvider"
                             predicateClassName="com.virtuslab.gitmachete.frontend.ui.impl.root.GitMacheteVisibilityPredicate"/>
-        <projectService serviceImplementation="com.virtuslab.gitmachete.frontend.actions.common.GitFetchSupportImpl"/>
 
         <fileType language="TEXT" fileNames="machete" name="MACHETE_FILE"
                   implementationClass="com.virtuslab.gitmachete.frontend.file.MacheteFileType" fieldName="INSTANCE"/>
         <lang.fileViewProviderFactory language="TEXT"
                                       implementationClass="com.virtuslab.gitmachete.frontend.file.MacheteFileViewProviderFactory"/>
         <completion.contributor language="TEXT"
                                 implementationClass="com.virtuslab.gitmachete.frontend.file.MacheteCompletionContributor"/>
+        <externalSystemManager implementation="com.virtuslab.gitmachete.frontend.externalSystem.MacheteProjectService"/>
+        <!-- "externalIconProvider" is available since 2020.1. The following line shall be uncommented after 2019.3 support is dropped. -->
+<!--        <externalIconProvider key="MACHETE" implementationClass="com.virtuslab.gitmachete.frontend.externalSystem.MacheteIconProvider"/>-->
 
+        <registryKey key="MACHETE.system.in.process" defaultValue="true"
+                     description="Whether IDEA should use 'in-process' mode for interaction with machete api"/>
```

Comment:
Hmmm a bit unclear what that means in practice :thinking:

Reply:
This is required to run refresh within IDE. Changing this value will cause an `UnsupportedOperationException`. Gradle handles it the same way. 

Bear in mind that registry window has this header:
![image](https://user-images.githubusercontent.com/19799111/85001838-749c7000-b154-11ea-84f1-c15ba36aeaae.png)

What `in-process` means:
![image](https://user-images.githubusercontent.com/19799111/85001265-bb3d9a80-b153-11ea-8cae-ed7cd1157f0b.png)

Comment wrt `isInProcessMode` dropped.

Reply:
`slave process` :scream: